docker-compose-rule-core/src/main/java/com/palantir/docker/compose/execution/DefaultDockerCompose.java

```diff
@@ -244,9 +263,20 @@ private static boolean downCommandWasPresent(String output) {
         return !output.contains("No such command");
     }
 
-    private String psOutput(String service) throws IOException, InterruptedException {
-        String psOutput = command.execute(Command.throwingOnError(), "ps", service);
-        validState(!Strings.isNullOrEmpty(psOutput), "No container with name '" + service + "' found");
-        return psOutput;
+    private static ErrorHandler swallowingServiceDoesNotExist() {
+        return (exitCode, output, commandName, commands) -> {
+            if (exitCode == 1 && output.isEmpty()) {
+                // Note: This (badly) checks if the command most likely failed due to
+                // the service not existing. If the ErrorHandler had access to
+                // error output, the proper way to check this would be to inspect
+                // the error output for "ERROR: No such service: <service_name>"
+
+                String fullCommand = Joiner.on(" ").join(Lists.asList(commandName, commands));
+                String serviceName = commands[commands.length - 1];
+
+                log.warn("It looks like `{}` failed.", fullCommand);
+                log.warn("This probably happened because no `{}` service exists.", serviceName);
```

Comment:
Shouldn't this throw rather than silently continue? I imagine we can't recover from this?

Reply:
If that output is produced, the command will have failed with empty output and then will trigger an exception right after it here: https://github.com/palantir/docker-compose-rule/pull/418/files#diff-cf7a91b7706a65a0c4cc650e69cf5d81R242

I agree that maybe a little misleading and we could clarify how `swallowingServiceDoesNotExist` handles the case